src/lib/y2sap/products/read_parameter.rb

```diff
@@ -86,30 +88,33 @@ def eval_product_ay
       Wizard.SetContents(
         _("Collecting installation profile for SAP product"),
         VBox(
-            Top(Left(Label(_("Please follow the on-screen instructions of SAP installer (external program)."))))
+          Top(
+            Left(
+              Label(
+                _("Please follow the on-screen instructions of SAP installer (external program).")
+              )
+            )
+          )
         ),
         "",
         true,
         true
       )
       Wizard.RestoreAbortButton()
       # First we execute the autoyast xml file of the product if this exeists
-      xml_path = get_product_parameter("ay_xml") == "" ? "" : @media.ay_dir_base + "/" +  get_product_parameter("ay_xml")
-      if @product_name == "B1"
-              SCR.Execute(path(".target.bash"), @media.sapinst_path + "/b1_hana_list.sh " + @media.inst_dir + " " + @media.ay_dir_base )
-      end
-      if File.exist?( xml_path )
-        SCR.Execute(path(".target.bash"), "sed -i s/##VirtualHostname##/" + @my_hostname + "/g " + xml_path )
-	#WFM.CallFunction("ayast_setup", ["setup","filename="+xml_path, "dopackages=yes" ] )
-        ret = openFile({ "filename" => xml_path, "dopackages" => "yes" })
-        log.info("ayast_setup returned for: " + xml_path)
-        if File.exist?("/tmp/ay_q_sid")
-           @sid = IO.read("/tmp/ay_q_sid").chomp
-        end
-        if File.exist?("/tmp/ay_q_sapinstnr")
-           @inst_number = IO.read("/tmp/ay_q_sapinstnr").chomp
-        end
-        SCR.Execute(path(".target.bash"), "mv /tmp/ay_* " + @media.inst_dir )
+      xml_path = get_product_parameter("ay_xml") == "" ? "" : @media.ay_dir_base + "/" + get_product_parameter("ay_xml")
+      SCR.Execute(
+        path(".target.bash"),
+        @media.sapinst_path + "/b1_hana_list.sh " + @media.inst_dir + " " + @media.ay_dir_base
```

Comment:
What do you mean here?

Reply:
When passing variables into a shell command you should always escape them to avoid problems with special characters (`&`, `>`, `;`, ...) or spaces. Esp. when the values come from outside (from user).

You can skip it when using safe hardcoded paths,  like that `b1_hana_list.sh`.